image_tools_py/src/showimage_py.py

```diff
@@ -0,0 +1,110 @@
+# Copyright 2017 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# System imports
+import sys
+import argparse
+
+# ROS2 imports
+import rclpy
+import sensor_msgs.msg
+
+# OpenCV imports
+import cv2
+
+import numpy
+
+# Local imports
+import image_qos_profile
+
+
+def encoding2mat(encoding):
+    encodings = {'mono': 1, 'bgr': 3, 'rgba': 4}
+    for prefix, channels in encodings.items():
+        if encoding.startswith(prefix):
+            break
+    else:
+        raise Exception("Unsupported encoding " + encoding)
+
+    types = {'8': 'uint8', '16': 'int16'}
+    for prefix, dtype in types.items():
+        if encoding[channels:] == prefix:
+            break
+    else:
+        raise Exception("Unsupported encoding " + encoding)
+
+    return dtype, channels
+
+
+def main():
+    # Pass command-line arguments to rclpy.
+    rclpy.init(sys.argv)
+
+    # Parse the command-line options.
+    parser = argparse.ArgumentParser(formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument(
+        '-d', '--depth', dest='queue_depth', action='store', default=10, type=int,
+        help='Queue depth')
+    parser.add_argument(
+        '-k', '--keep', dest='keep', action='store', default=1, type=int, choices=[0, 1],
+        help='History QoS setting, 0 - keep last sample, 1 - keep all the samples')
+    parser.add_argument(
+        '-r', '--reliability', dest='reliability_policy', action='store', default=1, type=int,
+        choices=[0, 1],
+        help='Reliability QoS setting, 0 - best effort, 1 - reliable')
+    parser.add_argument(
+        '-s', '--show', dest='show_camera', action='store', default=0, type=int, choices=[0, 1],
+        help='Show the camera stream')
+    args = parser.parse_args()
+
+    if args.show_camera == 1:
+        cv2.namedWindow('showimagepy')
```

Comment:
to be consistent with the executable names I would use `showimage_py` same below (same for cam2image)